azure/scope/cluster.go

```diff
@@ -87,6 +88,10 @@ func NewClusterScope(ctx context.Context, params ClusterScopeParams) (*ClusterSc
 		}
 	}
 
+	if params.Cache == nil {
+		return nil, errors.New("failed to generate new scope from nil Cache")
```

Comment:
nit: `failed to generate new scope due to nil Cache`

Reply:
no longer returning error for nil Cache